src/cursor.rs

```diff
@@ -501,4 +503,20 @@ mod tests {
         assert_eq!(x, saved_x);
         assert_eq!(y, saved_y);
     }
+
+    #[test]
+    fn test_set_cursor_style() {
+        assert!(SetCursorStyle::DefaultUserShape == SetCursorStyle::DefaultUserShape);
+        assert!(SetCursorStyle::BlinkingBlock != SetCursorStyle::BlinkingUnderScore);
+        assert_eq!(SetCursorStyle::BlinkingBlock, SetCursorStyle::BlinkingBlock);
+        assert_eq!(format!("{:?}", SetCursorStyle::SteadyBar), "SteadyBar");
+        let s1 = SetCursorStyle::DefaultUserShape;
+        let s2 = s1;
+        let s3 = s1.clone();
+        assert!(s2 == s1);
+        assert!(s3 == s1);
+        let mut hasher = DefaultHasher::new();
+        assert!(s1.hash(&mut hasher) == s2.hash(&mut hasher));
+        assert!(s1.hash(&mut hasher) == s3.hash(&mut hasher));
+    }
```

Comment:
I think I wouldn't worry about testing the derived traits unless there's a need to maintain some sort of backwards compatibility or external guarantee that these are a particular way. Thoughts on removing this?

```suggestion
```

Reply:
(I'd merge this with these tests btw if you feel strongly about it, it's just not code that I'd personally want to have lying around a project unless they're necessary)